docs/installation/keycloak/keycloak-minikube.md

```diff
@@ -0,0 +1,178 @@
+# How to Set Up Headlamp in a Minikube Cluster with Keycloak OIDC Authentication
+
+In this tutorial, we'll walk through the process of configuring Headlamp within a Minikube cluster while utilizing Keycloak for OIDC (OpenID Connect) authentication. This tutorial is based on Keycloak version 22.0.4, Minikube version v1.31.2, and Headlamp version 0.20.1.
+
+## Creating a Keycloak Client
+
+To enable OIDC authentication in your Minikube cluster, you'll need to create a Keycloak client. Follow these steps to configure the client in your Keycloak admin panel:
+
+1. Start by accessing your Keycloak admin panel.
+![Keycloak Admin Panel](./keycloak-admin-panel.jpeg)
+
+2. Navigate to the "Clients" section and Click on the "Create client" option.
+![Keycloak Clients Page](./keycloak-clients-list.jpeg)
+
+3. Fill in the "client-id" and "Name" fields and proceed to the next step.
+![Keycloak Create Client](./keycloak-create-client1.jpeg)
+
+4. In the "Capability Config" step, enable "Client authentication" and proceed to the next step.
+![Keycloak Create Client](./keycloak-create-client2.jpeg)
+
+
+6. Add "http://localhost:8000/*" to the "Valid redirect URIs" and save your settings.
+![Keycloak Create Client](./Keycloak-create-client3.jpeg)
+
+
+## Setting up Minikube with the Keycloak OIDC Configuration
+
+To configure Minikube for Keycloak OIDC integration, follow these steps:
+
+1. **Run the following command** to start Minikube with the necessary configuration options:
+
+   ```shell
+   minikube start -p=keycloak \
```

Comment:
I think you can use host.minikube.internal. [Refer](https://minikube.sigs.k8s.io/docs/handbook/host-access/)

Reply:
(I wrote a detailed further comment yesterday which seems to have been swallowed by the internets and fed into the daemon mouth of a /dev/null somewhere)

The tldr; is that WSL + Ubuntu + local windows browser doesn't work.

- the keycloak docs for kubernetes don't work because the yaml has VERSION in the container image name.
- the issuer url needs to be https, otherwise it doesn't work (there's a note about this in minikube and k8s docs).
- the keycloak docs for docker don't work because running keycloak this way support https (which is a requirement for k8s).
- minikube tunnel, and ingress don't work well with https because docker can't forward things properly to windows side with https (http can work).

Reply:
I used caddy to handles https, but I ran it in a server and did DNS config. I think you can do the same locally. Ill share the config that I used here after testing it locally.

Reply:
Cool. Caddy works with that config you sent, and minikube was able to start.